test/filter.spec.js

```diff
@@ -1144,5 +1144,32 @@ describe('resin-filter', () => {
 
       expect(sieve.filter(collection, input)).to.have.length(2)
     })
+
+    it('should behave correctly rule references a non-existent schema field', function () {
+      const schema = { test: { type: 'Short Text' } }
+      const collection = [
+        {
+          test: 'abcde',
+          id: 1
+        },
+        {
+          test: 'fghij',
+          id: 2
+        }
+      ]
+
+      const inputs = sieve.makeFilterInputs(schema)
+      const input = inputs.test
+
+      // Set the input name to something that doesn't exist in the schema
+      input.name = 'foobar'
+
+      input.operator = 'contains'
+      input.value = 'abc'
+
+      console.log(input)
+
+      expect(sieve.filter(collection, input)).to.have.length(0)
```

Comment:
When trying `npm test` this didn't fail on the old source!?!? Can you confirm this?

Reply:
@thgreasi Yep this is correct, the test here is more of a safegaurd against future regressions

Reply:
Would also love to have a test case that would fail without this PRs changes as a safegaurd against future regressions.

Reply:
The filtering logic actually behaves correctly in these situations, its the UI that fails :/
We need integration tests here ideally.

Reply:
I've created an issue for this https://github.com/resin-io-modules/resin-components/issues/254